x-pack/plugins/endpoint/server/plugin.ts

```diff
@@ -85,10 +82,18 @@ export class EndpointPlugin
     registerIndexPatternRoute(router, endpointContext);
   }
 
-  public start() {
+  public start(core: CoreStart, plugins: EndpointPluginStartDependencies) {
     this.logger.debug('Starting plugin');
+    endpointAppContextServices.start({
```

Comment:
Instead of adding `indexPatternRetriever` and `agentService` to a global variable, why not just pass it to the code that needs it.